src/result.rs

```diff
@@ -106,6 +133,14 @@ impl PartialEq for IonError {
     }
 }
 
+pub fn incomplete_data_error<T>(label: &'static str, offset: usize) -> IonResult<T> {
+    Err(incomplete_data_error_raw(label, offset))
+}
+
+pub fn incomplete_data_error_raw(label: &'static str, offset: usize) -> IonError {
```

Comment:
Why not `#[inline(never)]` this raw error function as well similar to other raw error functions.

Reply:
Great question! The other error methods each allocate a `String` via `format!`, which causes their assembly to include code for allocation, formatting, and deallocation. `incomplete_data_error_raw` holds a `&'static str`, so it doesn't need to allocate; its assembly is pretty small for now.

Reply:
Here's an assembly dump for `decoding_error_raw` (which is behind `inline(never)`):

![image](https://user-images.githubusercontent.com/611616/176208792-bd7d5ee3-363a-478d-abb3-c26eb96b8d07.png)

In contrast, here's the assembly for `incomplete_data_error_raw`, which usually gets inlined:

![image](https://user-images.githubusercontent.com/611616/176209513-875383a1-1635-411e-a18d-131aa6391a77.png)

I'm using [`cargo asm`](https://github.com/gnzlbg/cargo-asm) to generate these.